src/modules/profile/index.ts

```diff
@@ -1,15 +1,205 @@
 import { Profile } from 'dcl-catalyst-client/dist/client/specs/catalyst.schemas'
+import { getCatalystServersFromCache } from 'dcl-catalyst-client/dist/contracts-snapshots'
+import { AuthIdentity, Authenticator } from '@dcl/crypto'
+import { Entity, EntityType } from '@dcl/schemas'
 import { config } from '../config'
 
+export interface ConsistencyResult {
+  isConsistent: boolean
+  entity?: Entity
+  error?: string
+}
+
 export async function fetchProfile(address: string): Promise<Profile | null> {
   const PEER_URL = config.get('PEER_URL')
   const client = createLambdasClient({ url: PEER_URL + '/lambdas', fetcher: createFetchComponent() })
   try {
-    const profile = await client.getAvatarDetails(address)
+    const profile: Profile = await client.getAvatarDetails(address)
     return profile
   } catch (error) {
     return null
   }
 }
+
+export async function fetchProfileWithConsistencyCheck(address: string): Promise<ConsistencyResult> {
+  try {
+    // Determine network based on environment
+    const environment = config.get('ENVIRONMENT')
+    const network = environment === 'development' ? 'sepolia' : 'mainnet'
+
+    // Create content client for consistency check
+    const PEER_URL = config.get('PEER_URL')
+    const client = createContentClient({
+      url: PEER_URL + '/content',
+      fetcher: createFetchComponent()
+    })
+
+    // Get all catalyst servers for the network and remove the current peer to avoid duplicated fetches
+    const catalystServers = getCatalystServersFromCache(network)
+    const catalystUrls = catalystServers
+      .map(server => `${server.address}/content`)
+      .filter(url => url.toLowerCase() !== `${PEER_URL}/content`.toLowerCase())
+
+    // Check pointer consistency across all catalysts
+    const consistencyResult = await client.checkPointerConsistency(address, {
+      parallel: {
+        urls: catalystUrls
+      }
+    })
+
+    if (!consistencyResult.isConsistent) {
+      // Check if we have any valid entity from inconsistent catalysts
+      // This entity can be used for redeployment to fix the inconsistency
+      const validEntity =
+        consistencyResult.upToDateEntities && consistencyResult.upToDateEntities.length > 0
+          ? consistencyResult.upToDateEntities[0]
+          : undefined
+
+      return {
+        isConsistent: false,
+        entity: validEntity, // Entity for potential redeployment
+        error: 'Profile is not consistent across catalysts'
+      }
+    }
+
+    // If consistent and we have up-to-date entities, return the entity
+    if (consistencyResult.upToDateEntities && consistencyResult.upToDateEntities.length > 0) {
+      const entity = consistencyResult.upToDateEntities[0]
+      return {
+        isConsistent: true,
+        entity: entity
+      }
+    }
+
+    // If no entities found, profile doesn't exist (entity will be undefined)
+    return {
+      isConsistent: true,
+      entity: undefined
+    }
```

Comment:
What Do you think about refactoring this to a simplest way, something like

```suggestion
const { isConsistent, upToDateEntities = [] } = consistencyResult
const entity = upToDateEntities[0]

if (!isConsistent) {
  return {
    isConsistent: false,
    entity, 
    error: 'Profile is not consistent across catalysts'
  }
}

return {
  isConsistent: true,
  entity // could be undefined if no up-to-date entities found
}
```